### ISSUE TITLE
Add sort by last modified

### ISSUE THREAD

**Issue by dhowe** [nextcloud/contacts#272]
also a default sort that is perhaps more intuitive (is the current default sort documented somewhere?)

**Comment by skjnldsv** [comment]
In the settings 
![capture d ecran_2017-07-27_18-15-37](https://user-images.githubusercontent.com/14975046/28680769-9efc96b6-72f7-11e7-8676-4b823f79b6c7.png)

**Comment by olantrust** [comment]
Please include 'Date Created', and 'Last Modified' as sorting options.

**Comment by dhowe** [comment]
I don't see that option in my settings: 

![image](https://user-images.githubusercontent.com/737638/28681094-52ffb7ce-72c6-11e7-884f-b7afc17928e3.png)

**Comment by skjnldsv** [comment]
@dhowe You're using the latest official version, the current master has more than 200 commits. We're about to publish a new release! :)

**Comment by dhowe** [comment]
Excellent: do I have to download it, or is there an auto-update  ?

**Comment by skjnldsv** [comment]
Will be available as usual on the store.

**Comment by MorrisJobke** [comment]
It doesn't look like there is an easy way to determine when a contact was created or updated:

```xml
<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:cal="urn:ietf:params:xml:ns:caldav" xmlns:cs="http://calendarserver.org/ns/" xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns">
	<d:response>
		<d:href>/server/remote.php/dav/addressbooks/users/admin/contacts/8ee94958-71e2-4856-bf98-ebd335a13421.vcf</d:href>
		<d:propstat>
			<d:prop>
				<d:getetag>&quot;b8e32fc2f52f22e035176efb9345d5e5&quot;</d:getetag>
				<card:address-data>
					BEGIN:VCARD&#13;
					VERSION:3.0&#13;
					FN:New csdfgontact&#13;
					UID:8ee94958-71e2-4856-bf98-ebd335a13421&#13;
					REV:20170808T141227Z&#13;
					END:VCARD
				</card:address-data>
			</d:prop>
			<d:status>HTTP/1.1 200 OK</d:status>
		</d:propstat>
	</d:response>
</d:multistatus>
```

@georgehrke Or am I wrong?

**Comment by georgehrke** [comment]
@MorrisJobke wrong ticket?

**Comment by georgehrke** [comment]
To answer your question anyway: see http://sabre.io/dav/building-a-carddav-client/#finding-out-if-anything-changed

tl;dr: You have to keep a list of the etags.

**Comment by jonatoni** [comment]
@georgehrke he asked you because I'm trying to create a last modified sort option.

**Comment by skjnldsv** [comment]
@jonatoni isn't it the purpose of the REV field?

**Comment by georgehrke** [comment]
Yes, `REV` seems to be the way to go: https://tools.ietf.org/html/rfc6350#section-6.7.4

> In addition, the "REV" type described in Section 6.7.4 can be specified to indicate the last time that the vCard data was updated.

**Comment by mburnicki** [comment]
It would be very much appreciated it the sort order would ont only be "first name" or "last name" but in fact "last name, first name", across *all* address books, as it was available in the old owncloud contacts app.

I've recently upgraded from ownloud 8.2.x to nextcloud 12, and even though this new app looks much smarter, this is one of the features that I'm missing in contacts 1.5.3.

**Comment by olantrust** [comment]
**Location sort order**

Having a sort order of 'Location' would be amazing and very useful.
It could sort by the following priority:
Home Address
County > State > Town (if avail)
If no Home Address then
Work Address
Country > State > Town (if avail)
If no address then normal name sort order for the remainder.

**Comment by jancborchardt** [comment]
Sorry, I only saw this now – @skjnldsv @MorrisJobke @jonatoni what’s the use-case of sorting by anything else than First name, or Last name? Does any other contacts app have a »last modified« sort option?

What would be a nice sort could be connecting from the mail app to sort it by »Most contacted«, or also sort favorites up top as described in #105.

**Comment by skjnldsv** [comment]
The REV field is the `last modified` you're looking for! :)
![giphy](https://user-images.githubusercontent.com/14975046/30379248-1df102ea-9896-11e7-8de7-eec014e04565.gif)

**Comment by mburnicki** [comment]
@jancborchardt Eventually to see which contacts have recently been modified, e.g. uploaded from a smartphone after a new phone number has been added, or something. I've seen quite some times that vcards have been messed up in such cases, so it could be very useful to check the latest changes.
Concerning sorting by lastname or firstname: As mentioned earlier I'd find it very useful to really sort by "Lastname, Firstname" to have a proper sort order even if you have a huge number of family members with the same lastname, and it would be even more overseeable if this was reflected in the side bar on the left.

**Comment by olantrust** [comment]
Last modified would be incredibly useful.
People add contacts on-the-go at meetings or parties, via their smartphone.
They will want to either remember who they are, or add extra information at a later stage.
Last modified makes that much easier.

**Comment by jancborchardt** [comment]
Alright, recently modified seems to make sense. However not a priority since it’s a big overhead and also there doesn’t seem anything comparable in common apps.

> Sorting by location/country would be incredibly useful for obvious reasons (when people are visiting or dealing with a country or city and want a quick view of who to contact/visit).

We have a search for this, which I also use for exactly this case. ;)

**Comment by Henni** [comment]
@olantrust *recently created* might be a good application for "smart groups".
I believe there already exists an issue regarding smart groups.

**Comment by alexanderdd** [comment]
Sort by date created would actually be very helpful, also in a business context where you could see new customers at the top.

**Comment by skjnldsv** [comment]
Sorting by date pull request is here: #992 :tada: 
Please come and help review!
